packages/frontend/src/hooks/useFieldValues.ts

```diff
@@ -96,7 +99,6 @@ export const useFieldValues = (
             ),
         {
             // make sure we don't cache for too long
-            staleTime: 60 * 1000, // 1 minute
```

Comment:
Not entirely sure why this was not working. 
If I searched, went back to the tables page, go again the to table and made the same search, useQuery would not trigger the `onSuccess` with the cached data.

Reply:
apparently react-query doesn't refetch a stale query on every render (ref: https://stackoverflow.com/a/72992274)

Reply:
Yup, but I expected the `onSuccess` to be triggered with the cached results.
But I suppose what happens is they return the cached data as in `const {data} = useQuery(...)` without triggering the `onSuccess`. So this means we should move all the logic from onSuccess to `useEffect(..., [data]);`

Reply:
yup, exactly! Guess it aligns with them deprecating `onSuccess` and other callbacks in most recent versions 👍

Reply:
Decided that this would involve a bit of a refactor and we would need to retest everything. 
Going to merge the current fix without state time.